File: Data Structures/Hash/KTimesInArray.py
def findK(arr, n, k):

    count_map = {}
    calc_k = 0
    for num in arr:
        if num in count_map:
            count_map[num] += 1
            if count_map[num] is k:
                return num
        else:
            count_map[num] = 1
    return -1

arr = [1,1,2,3,4]
value = findK(arr, len(arr), 2)
print(value)

File: Data Structures/Hash/KTimesInArray.java
import java.util.HashMap;

class KTimesInArray
{
    static int countK(int arr[], int n, int k)
    {
        HashMap<Integer, Integer> count_map = new HashMap<Integer, Integer>();
        for (int i = 0; i < n; i++)
        {
            if(count_map.get(arr[i]) != null)
            {
                count_map.put(arr[i], count_map.get(arr[i])+1);
                if(count_map.get(arr[i]) == k)
                    return arr[i];
            }
            else
            {
                count_map.put(arr[i], 1);
            }
        }
        return -1;
    }

    public static void main(String args[])
    {
        int arr[] = {1,7,4,3,4,8,7};
        int n = arr.length;
        int k =2;
        System.out.println(countK(arr, n, k));
    }
}
